l293x/l293x.go

```diff
@@ -56,49 +56,71 @@ func (d *Device) Stop() {
 	d.en.Low()
 }
 
+// PWM is the interface necessary for controlling the motor driver.
+type PWM interface {
+	Configure(config machine.PWMConfig) error
+	Channel(pin machine.Pin) (channel uint8, err error)
+	Top() uint32
+	Set(channel uint8, value uint32)
+	SetPeriod(period uint64) error
+}
+
```

Comment:
It might be better to move `type PWM interface` to `./pwm.go` .

Reply:
No, I think it's better to keep it in the driver. The reason is that (unlike SPI or I2C), the capabilities of each PWM vary. For example: the tone driver (https://github.com/tinygo-org/drivers/pull/251) requires a `SetPeriod` method while the servo driver (https://github.com/tinygo-org/drivers/compare/dev...servo) doesn't need this method. If there is a PWM that does not support the `SetPeriod` method, it can still be used for a servo. If the `PWM` interface is shared between all drivers, such a PWM (without SetPeriod) would not be usable by any driver.

That said, most of the methods here are unused so might be removed from this interface (for example, `SetPeriod` is unused).

Reply:
I understood.

Reply:
Also, at least for now, none of the interfaces defined in this package have any reference to the `machine` package. This is what makes it possible to run the unit tests just using "standard" Go.